src/pyk/cterm.py

```diff
@@ -40,23 +40,43 @@ class CTerm:
     constraints: tuple[KInner, ...]
 
     def __init__(self, config: KInner, constraints: Iterable[KInner] = ()) -> None:
-        self._check_config(config)
-        constraints = self._normalize_constraints(constraints)
+        if CTerm._is_top(config):
+            config = mlTop()
+            constraints = ()
+        elif CTerm._is_bottom(config):
+            config = mlBottom()
+            constraints = ()
+        else:
+            self._check_config(config)
+            constraints = self._normalize_constraints(constraints)
         object.__setattr__(self, 'config', config)
         object.__setattr__(self, 'constraints', constraints)
 
     @staticmethod
     def from_kast(kast: KInner) -> CTerm:
-        config, constraint = split_config_and_constraints(kast)
-        constraints = flatten_label('#And', constraint)
-        return CTerm(config, constraints)
+        if CTerm._is_top(kast):
+            return CTerm.cterm_top()
+        elif CTerm._is_bottom(kast):
+            return CTerm.cterm_bottom()
+        else:
+            config, constraint = split_config_and_constraints(kast)
+            constraints = flatten_label('#And', constraint)
+            return CTerm(config, constraints)
 
     @staticmethod
     def from_dict(dct: dict[str, Any]) -> CTerm:
         config = KInner.from_dict(dct['config'])
         constraints = [KInner.from_dict(c) for c in dct['constraints']]
         return CTerm(config, constraints)
 
+    @staticmethod
+    def cterm_top() -> CTerm:
+        return CTerm(mlTop(), ())
+
+    @staticmethod
+    def cterm_bottom() -> CTerm:
+        return CTerm(mlBottom(), ())
+
```

Comment:
```suggestion
    @staticmethod
    def top() -> CTerm:
        return CTerm(mlTop(), ())

    @staticmethod
    def bottom() -> CTerm:
        return CTerm(mlBottom(), ())

```

I guess this will let us write `CTerm.top()` instead of `CTerm.cterm_top()`, which feels redundant.